ibis/expr/types/groupby.py

```diff
@@ -250,33 +246,24 @@ def _selectables(self, *exprs, **kwexprs):
         [`GroupedTable.mutate`](#ibis.expr.types.groupby.GroupedTable.mutate)
         """
         table = self.table
-        default_frame = self._get_window()
+        default_frame = ops.RowsWindowFrame(
+            table=self.table,
+            group_by=bind_expr(self.table, self.by),
+            order_by=bind_expr(self.table, self._order_by),
+        )
         return [
-            an.windowize_function(e2, frame=default_frame)
+            an.windowize_function(e2, default_frame)
             for expr in exprs
             for e1 in util.promote_list(expr)
             for e2 in util.promote_list(table._ensure_expr(e1))
         ] + [
-            an.windowize_function(e, frame=default_frame).name(k)
+            an.windowize_function(e, default_frame).name(k)
             for k, expr in kwexprs.items()
             for e in util.promote_list(table._ensure_expr(expr))
         ]
 
     projection = select
 
-    def _get_window(self):
-        if self._window is None:
-            return ops.RowsWindowFrame(
-                table=self.table,
-                group_by=self.by,
-                order_by=bind_expr(self.table, self._order_by),
-            )
-        else:
-            return self._window.copy(
-                groupy_by=bind_expr(self.table, self._window.group_by + self.by),
-                order_by=bind_expr(self.table, self._window.order_by + self._order_by),
-            )
```

Comment:
Nice, this was pretty gross.

Reply:
The whole GroupedTable object must be refactor, subject to a follow-up.